### ISSUE TITLE
pandas row_condition causes an error when edit existing suite if simple quote in query expression

### ISSUE THREAD

**Issue by benoitLebreton-perso** [fivetran/great_expectations#2536]
**Describe the bug**

When I use the `row_condition=<pandas query>` option there is no problem in the moment. And this is a very convenient feature so bravo !
The little problem is I can have an issue later when I want to modify my already existing expectation suite.
This issue will appear only if I code the pandas query 
`row_condition="domain == 'product' "` : with simple quote inside
and not if 
`row_condition='domain == "product" '` : with double quote inside

**To reproduce**

I already have a suite and it contains at least one expectation with a row_condition with simple quote inside. 
```
batch.expect_column_value_lengths_to_be_between(
    column="title", 
    min_value=1, 
    max_value=100, 
    row_condition="domain == 'product'",
    condition_parser='pandas')
```
I want to edit it again. 

`$ great_expectations suite edit <my-suite-name>`

I have the following error : 

`black.InvalidInput: Cannot parse: 1:145: batch.expect_column_value_lengths_to_be_between(column='title', condition_parser='pandas', max_value=100, min_value=1, row_condition='domain == 'product'')`

If I anticipate and declare`row_condition='domain == "product"'` instead (with double quotes inside !) Then there is no error :)
Maybe it is obvious but the problem is pandas let us choose if we put single or double inside the expression. So I think we could prevent this problem.

**What could we do ?

A/ We could tell everybody to use double quote inside the pandas query. In the documentation for instance.
B/ We could raise an error when the user declares an row_condition with a simple quote (preventive, preferable in my opinion)
C/ We could modify the row_condition parser to let it parse simple quote (curative, more difficult in my opinion)

My advise would be to the A+B solution, because it is simple and would avoid all future error.
The C would correct even the reading of already existing suite but because the row_condition feature is experimental, I think B is more reasonable.

**Comment by eugmandel** [comment]
@benoitLebreton-perso Thank you for reporting this! I agree - A+B sounds like a relatively quick and acceptable way to address this issue. I will label this issue with "help wanted" - this is our way of asking for a community PR.

**Comment by talagluck** [comment]
Resolved by #3676
